### ISSUE TITLE
`max_treedepth` warning in `pm.sample()`?

### ISSUE THREAD

**Issue by larryshamalama** [pymc-devs/pymc#6587]
### Description

Running the following simple model:

```python
import pymc as pm

with pm.Model() as model:
    mu = pm.Normal("mu", 0., 5.)
    x = pm.Normal("x", mu, 3.)
    
    trace = pm.sample()
```

yields the following warning message:

```
Sampling 4 chains for 1_000 tune and 1_000 draw iterations (4_000 + 4_000 draws total) took 1 seconds.
Chain <xarray.DataArray 'chain' ()>
array(0)
Coordinates:
    chain    int64 0 reached the maximum tree depth. Increase `max_treedepth`, increase `target_accept` or reparameterize.
Chain <xarray.DataArray 'chain' ()>
array(1)
Coordinates:
    chain    int64 1 reached the maximum tree depth. Increase `max_treedepth`, increase `target_accept` or reparameterize.
Chain <xarray.DataArray 'chain' ()>
array(2)
Coordinates:
    chain    int64 2 reached the maximum tree depth. Increase `max_treedepth`, increase `target_accept` or reparameterize.
Chain <xarray.DataArray 'chain' ()>
array(3)
Coordinates:
    chain    int64 3 reached the maximum tree depth. Increase `max_treedepth`, increase `target_accept` or reparameterize.
```

Is this behaviour due to a recent addition to the codebase?

**Comment by ricardoV94** [comment]
Yes, I think it is a bug. A user on discourse narrowed it down quite a lot: https://discourse.pymc.io/t/bug-in-warning-treedepth/11576/4

**Comment by ricardoV94** [comment]
CC @michaelosthege

**Comment by larryshamalama** [comment]
Oh, I didn't check Discourse, thanks for the pointer

**Comment by michaelosthege** [comment]
The output you posted above doesn't look like a warning to me - are you saying that this is a logging output at WARNING level?
If that's the case the log message should indicate where it originated from.

**Comment by larryshamalama** [comment]
> The output you posted above doesn't look like a warning to me - are you saying that this is a logging output at WARNING level? If that's the case the log message should indicate where it originated from.

Maybe, I'm not that familiar with the warnings and logging messages. The ones above seem to be generated from [`warn_divergences`](https://github.com/pymc-devs/pymc/blob/b6521f2d6725d4c8d2ed682c7017baae5ebf55a2/pymc/stats/convergence.py#L164-L171). I see that you wrote the function, so your assessment is probably more accurate than mine 😅

**Comment by michaelosthege** [comment]
> > The output you posted above doesn't look like a warning to me - are you saying that this is a logging output at WARNING level? If that's the case the log message should indicate where it originated from.
> 
> Maybe, I'm not that familiar with the warnings and logging messages. The ones above seem to be generated from [`warn_divergences`](https://github.com/pymc-devs/pymc/blob/b6521f2d6725d4c8d2ed682c7017baae5ebf55a2/pymc/stats/convergence.py#L164-L171). I see that you wrote the function, so your assessment is probably more accurate than mine 😅

Okay, so that [line 167](https://github.com/pymc-devs/pymc/blob/b6521f2d6725d4c8d2ed682c7017baae5ebf55a2/pymc/stats/convergence.py#L167) should be updated to do `int(c)` within that f-string, so we get a more concise output.

...looks like when I refactored that code (I didn't write the function) I unintentionally changed it to look at `tree_depth` instead of `reached_max_treedepth`.

👉👉👉 PR to fix this: #6591
